File: app/src/main/java/fr/gstraymond/ui/view/impl/TypePTView.java
package fr.gstraymond.ui.view.impl;

import android.widget.TextView;

import fr.gstraymond.R;
import fr.gstraymond.models.search.response.Card;
import fr.gstraymond.tools.PowerToughnessFormatter;
import fr.gstraymond.tools.TypeFormatter;
import fr.gstraymond.ui.view.CommonDisplayableView;

public class TypePTView extends CommonDisplayableView<TextView> {

    private PowerToughnessFormatter ptFormatter;
    private TypeFormatter typeFormatter;

    public TypePTView() {
        this.ptFormatter = new PowerToughnessFormatter();
        this.typeFormatter = new TypeFormatter();
    }

    @Override
    public void setValue(Card card, int position) {
        String ptOrType = getPtOrType(card);
        if (ptOrType.isEmpty()) getView().setText("");
        else getView().setText(" — " + ptOrType);
    }

    @Override
    public boolean display(Card card) {
        return super.display(true);
    }

    @Override
    public int getId() {
        return R.id.array_adapter_card_type_pt;
    }

    private String getPtOrType(Card card) {
        if (card.getLoyalty() != null) return card.getLoyalty();

        String pt = ptFormatter.format(card);
        if (!pt.isEmpty()) return pt;

        return typeFormatter.formatFirst(card);
    }
}

File: app/src/main/kotlin/fr/gstraymond/ui/view/impl/TypePTView.kt
package fr.gstraymond.ui.view.impl

import android.widget.TextView

import fr.gstraymond.R
import fr.gstraymond.models.search.response.Card
import fr.gstraymond.tools.PowerToughnessFormatter
import fr.gstraymond.tools.TypeFormatter
import fr.gstraymond.ui.view.CommonDisplayableView

class TypePTView : CommonDisplayableView<TextView>(R.id.array_adapter_card_type_pt) {

    private val ptFormatter = PowerToughnessFormatter()
    private val typeFormatter = TypeFormatter()

    override fun setValue(view: TextView, card: Card, position: Int) {
        val ptOrType = getPtOrType(card)
        if (ptOrType.isEmpty()) view.text = ""
        else view.text = " — $ptOrType"
    }

    override fun display(view: TextView, card: Card) = display(view, true)

    private fun getPtOrType(card: Card): String {
        val pt = ptFormatter.format(card)
        return when {
            card.loyalty != null -> card.loyalty
            pt.isNotEmpty() -> pt
            else -> typeFormatter.formatFirst(card)
        }
    }
}
